### PR TITLE
CDS4j: Update with expressions

### PR DESCRIPTION
**Pull request by MattSchur**

### CODE REVIEW

File: java/query-api.md [hunk 1]
```diff
@@ -986,6 +986,25 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
    .data("title", "CAP Matters");
 ```
 
+### Update with Expressions (beta) {#update-expressions}
+
+In addition to the update data with plain Java values, you can also update with CQL expressions via the `set` method, e.g. to decrease the stock of Book 101 by 1:
```
Comment (agoerler) [inline java/query-api.md:991]:
```suggestion
The [data](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#data(java.util.Map)), [entry](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entry(java.util.Map)), and  [entries](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entries(java.lang.Iterable)) methods allow to specify the new values as plain Java values. In addition or alternatively you can use the `set` method to specify the new value as a `CqnValue`, which can even be an expression. This allows, for example, to decrease the stock of Book 101 by 1:
```

**Review by agoerler — APPROVED** [review]
Looks good to me. A minor suggestion only.

**Review by agoerler — CHANGES_REQUESTED** [review]
We also need to document that the setter expressions are evaluated on the DB and the result of the expressions is not available in the `Result` of the execution of the `Update`.

The application might have to read after write in custom code.

**Comment by agoerler** [comment]
> Document result of update with expressions

https://github.com/cap-js/docs/pull/591/commits/6de14aede3f83b9bece0398175fe59f61dd2bc70

**Review by agoerler — APPROVED** [review]

File: java/query-api.md [hunk 2]
```diff
@@ -986,6 +986,25 @@ Update.entity(BOOKS, b -> b.matching(Books.create(100)))
    .data("title", "CAP Matters");
 ```
 
+### Update with Expressions (beta) {#update-expressions}
+
+The [data](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#data(java.util.Map)), [entry](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entry(java.util.Map)), and  [entries](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entries(java.lang.Iterable)) methods allow to specify the new values as plain Java values. In addition/alternatively you can use the `set` method to specify the new value as a `CqnValue`, which can even be an expression. This allows, for example, to decrease the stock of Book 101 by 1:
```
Comment (renejeglinsky) [inline java/query-api.md:991]: Can we add a capire link for CqnValue?

Comment (agoerler) [inline java/query-api.md:991]:
```suggestion
The [data](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#data(java.util.Map)), [entry](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entry(java.util.Map)), and  [entries](https://javadoc.io/doc/com.sap.cds/cds4j-api/latest/com/sap/cds/ql/Update.html#entries(java.lang.Iterable)) methods allow to specify the new values as plain Java values. In addition/alternatively you can use the `set` method to specify the new [value](#values) as a `CqnValue`, which can even be an expression. This allows, for example, to decrease the stock of Book 101 by 1:
```

**Review by renejeglinsky — APPROVED** [review]
Thanks!

**Pull request merged**